pkg/skaffold/deploy/status_check_test.go

```diff
@@ -0,0 +1,363 @@
+			actual, err := getDeployments(client, "test", labeller)
+			t.CheckErrorAndDeepEqual(test.shouldErr, err, test.expected, actual)
+		})
+	}
+}
+
+type MockRolloutStatus struct {
+	called    int
+	responses []string
+	err       error
+}
+
+func (m *MockRolloutStatus) Executefunc(context.Context, *kubectl.CLI, string) (string, error) {
+	var resp string
+	if m.err != nil {
+		m.called++
+		return "", m.err
+	}
+	if m.called >= len(m.responses) {
+		resp = m.responses[len(m.responses)-1]
+	} else {
+		resp = m.responses[m.called]
+	}
+	m.called++
+	return resp, m.err
+}
+
+func TestPollDeploymentRolloutStatus(t *testing.T) {
+
+	var tests = []struct {
+		description string
+		mock        *MockRolloutStatus
+		duration    int
+		exactCalls  int
+		shouldErr   bool
+	}{
+		{
+			description: "rollout returns success",
+			mock: &MockRolloutStatus{
+				responses: []string{"dep successfully rolled out"},
+			},
+			exactCalls: 1,
+			duration:   500,
+		},
+		{
+			description: "rollout returns error in the first attempt",
+			mock: &MockRolloutStatus{
+				err: errors.New("deployment.apps/dep could not be found"),
+			},
+			shouldErr:  true,
+			exactCalls: 1,
+			duration:   500,
+		},
+		{
+			description: "rollout returns success before time out",
+			mock: &MockRolloutStatus{
+				responses: []string{
+					"Waiting for rollout to finish: 0 of 1 updated replicas are available...",
+					"Waiting for rollout to finish: 0 of 1 updated replicas are available...",
+					"deployment.apps/dep successfully rolled out"},
+			},
+			duration:   500,
+			exactCalls: 3,
+		},
+		{
+			description: "rollout returns did not stabilize within the given timeout",
+			mock: &MockRolloutStatus{
+				responses: []string{
+					"Waiting for rollout to finish: 1 of 3 updated replicas are available...",
+					"Waiting for rollout to finish: 1 of 3 updated replicas are available...",
+					"Waiting for rollout to finish: 2 of 3 updated replicas are available..."},
+			},
+			duration:  1000,
+			shouldErr: true,
+		},
+	}
+	originalPollingPeriod := defaultPollPeriodInMilliseconds
+	for _, test := range tests {
+		testutil.Run(t, test.description, func(t *testutil.T) {
+			mock := test.mock
+			// Figure out why i can't use t.Override.
+			// Using t.Override throws an error "reflect: call of reflect.Value.Elem on func Value"
+			executeRolloutStatus = mock.Executefunc
+			defer func() { executeRolloutStatus = getRollOutStatus }()
+
+			defaultPollPeriodInMilliseconds = 100
+			defer func() { defaultPollPeriodInMilliseconds = originalPollingPeriod }()
+
+			actual := &sync.Map{}
+			pollDeploymentRolloutStatus(context.Background(), &kubectl.CLI{}, "dep", time.Duration(test.duration)*time.Millisecond, actual)
+
+			if _, ok := actual.Load("dep"); !ok {
+				t.Error("expected result for deployment dep. But found none")
+			}
+			err := getSkaffoldDeployStatus(actual)
+			t.CheckError(test.shouldErr, err)
+			if test.exactCalls > 0 {
```

Comment:
nit: don't need this check

Reply:
Actually we do, because this is a time sensitive test and poll duration is small for unit tests, sometime mock could get called n+1 times or n, or n-1 times when things timeout.
I added some description there.